code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java

```diff
@@ -381,4 +382,21 @@ protected void notConfigured() {
                 "NotConfigured", "Ldap is not configured", null);
     }
 
+    public List<Identity> getIdentities(List<Map<String, String>> identitiesGiven) {
+        if (identitiesGiven == null || identitiesGiven.isEmpty()){
+            return new ArrayList<>();
+        }
+
+        List<Identity> identities = new ArrayList<>();
+        for (Map<String, String> identity: identitiesGiven){
+            String externalId = identity.get(IdentityConstants.EXTERNAL_ID);
+            String externalIdType = identity.get(IdentityConstants.EXTERNAL_ID_TYPE);
+            Identity gotIdentity = getIdentity(externalId, externalIdType);
+            if (gotIdentity == null) {
+                throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "InvalidIdentity", "Invalid Identity", null);
+            }
+            identities.add(gotIdentity);
+        }
+        return identities;
+    }
```

Comment:
Moved this method from ADIdentityProvider to LDAPIdentityProvider.java - LDAPIdentityProvider.java is the base class for AD and OpenLDAP. Now both AD and OpenLDAP need this method.